### ISSUE TITLE
Automatic conversion from WinRT class to implemented interface

### ISSUE THREAD

**Issue by kaivol** [microsoft/windows-rs#2759]
### Use case
Currently, the following code does not compile:
```rust
let vector: Collections::IVector<IStringable> = ...;
let uri = Uri::CreateUri(h!("https://github.com"))?;
vector.Append(&uri)?;
       ------  ^^^ the trait `CanInto<IStringable>` is not implemented for `Uri`
       |
       required by a bound introduced by this call
= note: required for `&windows::Foundation::Uri` to implement `IntoParam<windows::Foundation::IStringable, ReferenceType>`
```
Instead, one has to manually cast the `Uri` to a `IStringable`. This should not be necessary, because  it is guaranteed that [`Uri` implements `IStringable`](https://learn.microsoft.com/en-us/uwp/api/windows.foundation.uri?view=winrt-22621).

I understand that `Uri` implements `CanTryInto<IStringable>`, which makes it compatible with `TryIntoParam<IStringable>` arguments, but `TryIntoParam` arguments only seem to be used for a few methods. 

The same problem also exists for WinRT interfaces which require another interface. 

### Suggestion
I'm not sure if I'm missing something, but I would suggest getting rid of the distinction between `CanInto`/`IntoParam` and `CanTryInto`/`TryIntoParam`, replacing the latter with the former. 
This would require changes to `CanInto` to support conversions using `QueryInterface`, but that should not be a problem, as the conversions are guaranteed to succeed. 

---
I would be happy to hear any opinion on this; whether the proposal makes sense or if there are reasons for not doing this and keeping the current solution.

**Comment by kennykerr** [comment]
It may be possible to combine `TryIntoParam` and `IntoParam` so long as we can do it without a run time cost to existing uses of `IntoParam`.

**Comment by kennykerr** [comment]
This should fix it: #2759
